### ISSUE TITLE
Import errors should include full path of the file that is being imported

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#11042]
## Description
When using relative paths the error message often (but not always) shows only a relative path to the file. When using path remapping and base path together with various flavors of relative imports (see #11036) it might not be obvious where the compiler is looking for the file. I think the compiler should display the final, resolved filesystem path instead of (or in addition to) the relative one.

### Example 1
`/tmp/a/b/contract.sol`
```solidity
import "relative.sol";
```
```bash
cd /tmp/c/d/
solc /tmp/a/b/contract.sol --allow-paths .
```
Here the compiler will look for the file in `/tmp/c/d/relative.sol`, not `/tmp/a/b/relative.sol` as some users might (wrongly) expect. It only says `relative.sol` though:
```
Error: Source "relative.sol" not found: File not found.
 --> /tmp/a/b/contract.sol:1:1:
  |
1 | import "relative.sol";
  | ^^
```

### Example 2
`/tmp/a/b/contract.sol`
```solidity
import "/tmp/absolute.sol";
```
```bash
cd /tmp/c/d/
solc /tmp/a/b/contract.sol /tmp=/contracts --base-path /home
```

Here the compiler says that the missing import is `/contracts/absolute.sol` but omits the fact that it prepends base path to it and actually expects the file at `/home/contracts/absolute.sol`:
```
Error: Source "/contracts/absolute.sol" not found: File not found.
 --> /tmp/a/b/contract.sol:1:1:
  |
1 | import "/tmp/absolute.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The fact that the base path is prepended to an absolute path is of course a bug (reported in #11036) but showing the full path would at least allow the user to realize why his code does not compile.

## Environment

- Compiler version: 0.8.1

**Comment by chriseth** [comment]
This is a simple change: Just add the path we were looking at in `CommandLienInterface::processInput()::fileReader` to the error message this function returns.

**Comment by chriseth** [comment]
Implemented in #12434
